net/unbound/files/unbound.sh

```diff
@@ -223,6 +231,135 @@ create_interface_dns() {
 
 ##############################################################################
 
+create_local_zone() {
+  local target="$1"
+  local partial domain found
+
+
+  if [ -n "$UNBOUND_LIST_DOMAINS" ] ; then
+    for domain in $UNBOUND_LIST_DOMAINS ; do
+      case $target in
+      *"${domain}")
+        found=1
+        break
+        ;;
+
+      [A-Za-z]*.[A-Za-z]*)
+        found=0
+        ;;
+
+      *) # no dots
+        found=1
+        break
+        ;;
+      esac
+    done
+  else
+    found=0
+  fi
+
+
+  if [ $found -eq 0 ] ; then
+    # New Zone! Bundle local-zones: by first two name tiers "abcd.tld."
+    partial=$( echo "$target" | awk -F. '{ j=NF ; i=j-1; print $i"."$j }' )
```

Comment:
`[ $found -eq 0 ]` will only occur if you have at least a dot with alpha-numeric on both sides and that domain is not yet in aggregated list variable. For each `local-data:` it helps Unbound to provide `local-zone:` and if we don't, then it will create a `local-zone:` of the complete FQDN. We only want the `local-zone:` once also.